crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -394,6 +396,10 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
             fn new_archetype(&mut self, archetype: &#path::archetype::Archetype, system_state: &mut #path::system::SystemState) {
                 self.state.new_archetype(archetype, system_state)
             }
+
+            fn default_config() -> TSystemParamState::Config {
+                TSystemParamState::default_config()
```

Comment:
Maybe add a comment that this reuses the tuple impls?

Personally, I'd change it to use a custom struct, because I'd rather be able to do
```rust
struct MyThing<'a>{ local: Local<'a, u32> };
fn my_sys(f: MyThing){}
my_sys.system().configure(|value| value.0.local=Some(u32));
```
instead of the current
```rust
my_sys.system().configure(|value| value.0.0 =Some(u32));
```

@cart is there a reason that this doesn't create a custom struct which is a copy of T?

As a bonus this would get correct privacy for free, as well as allowing infinitely large `derive(SystemParam)` structs.

Reply:
I would rather not change the api this close to the release, with the blog, the migration guide and the cheat book mostly written / updated

Reply:
That makes sense. I think that this current API is extremely ugly and I don't like the footgun of it only being limited to 16, and field ordering being significant

Id personally rather see it changed before 0.5, but this PR isn't the place to decide that.

Reply:
Yeah im open to considering alternatives, but probably not for 0.5. I generally just try to avoid creating new symbols via macros unless absolutely necessary. But it does feel necessary here / i wasn't properly considering the implications of this choice for the user-facing config api.